neurolib/models/model.py

```diff
@@ -474,25 +464,28 @@ def setOutput(self, name, data, append=False, removeICs=False):
                 raise ValueError(f"Don't know how to truncate data of shape {data.shape}.")
 
         # subsample to sampling dt
-        if data.ndim == 1:
-            data = data[:: self.sample_every]
-        elif data.ndim == 2:
-            data = data[:, :: self.sample_every]
-        else:
-            raise ValueError(f"Don't know how to subsample data of shape {data.shape}.")
+        if data.shape[-1] >= self.params["duration"] - self.startindt:
```

Comment:
Does this `if` clause prevent any previous case to run?

Reply:
This catches unintended subsampling of BOLD.